### PR TITLE
HPCC-14828 VS 2015 build has conflicting errno definitions

### PR DESCRIPTION
**Pull request by afishbeck**
jsocket redefines winsock errors to posix errors if errno.h is not
included.  In Visual Studio 2015 errno.h is included indirectly.

Will now create uniquely named defines for these errors that are mapped
either to winsock (windows) or errno.h (posix) depending on the system.

Signed-off-by: Anthony Fishbeck anthony.fishbeck@lexisnexis.com

### CODE REVIEW

**Comment by richardkchapman** [comment]
@mckellyln Please review

**Comment by mckellyln** [comment]
with VS2015 errno.h is, for example, EADDRINUSE not the same value as WSAEADDRINUSE ?

File: system/jlib/jsocket.cpp [hunk 1]
```diff
@@ -651,7 +662,7 @@ typedef union {
 #define DEFINE_SOCKADDR(name) J_SOCKADDR name; memset(&name,0,sizeof(J_SOCKADDR))
 
 
-#define EINTRCALL EINTR
+#define JSE_INTR EINTR
```
Comment (mckellyln) [inline system/jlib/jsocket.cpp]: I would make it JSE_EINTR to match posix name.

**Comment by afishbeck** [comment]
No, that's exactly the problem.  The values are all different.

**Comment by mckellyln** [comment]
ok, I think it looks fine.

**Pull request merged**